backend/api/interfaces/api_controllers/webhook/stripe.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any
+import json
+
+import stripe
+from api.common.utils import get_logger
+from api.core.container import (
+    get_stripe_setting_service,
+    get_billing_record_service
+)
+from api.usecases.stripe_setting_service import StripeSettingService
+from api.usecases.billing_record_service import BillingRecordService
+from fastapi import APIRouter, Depends, HTTPException, Request, status
+from fastapi.responses import JSONResponse
+
+logger = get_logger(__name__)
+
+router = APIRouter(prefix="/webhooks")
+router.tags = ["Stripe"]
+
+SUPPORTED_EVENTS = {
+    "invoice.paid": lambda obj, svc: handle_invoice_paid(obj, svc),
+}
+
+
+@router.post("/stripe", status_code=status.HTTP_200_OK)
+async def stripe_webhook(
+    request: Request,
+    stripe_setting_service: StripeSettingService = Depends(get_stripe_setting_service),
+    billing_record_service: BillingRecordService = Depends(get_billing_record_service),
+):
+    event = None
+    payload = await request.body()
+    stripe_settings = await stripe_setting_service.get_stripe_secret()
+
+    # If there is no Stripe settings. Return early
+    if not stripe_settings or not stripe_settings.stripe_webhook_secret:
```

Comment:
We must check tenant_id here as well. It is because as a host they can keep the strip settings at the environment level.. only tenants can update strip settings via UI to store it on their DB. 
```py
    if not stripe_settings or not stripe_settings.stripe_webhook_secret:
        if request.state.tenant_id:    
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail="Webhook secret not set on server",
            )
        else:
            # This will be consider as a host. 
            stripe_settings = None
```